### PR TITLE
Revert ConditionalClauseNavigatorEntry change and reimplement selection on clause label

### PR DESCRIPTION
**Pull request by gbalint**
**Description:**
It turned out that my change in https://github.com/concrete-utopia/utopia/pull/3480 was not a good idea.
In that PR, I changed the `ConditionalClauseNavigatorEntry.elementPath` to contain the element path of the root element of the clause. However, this caused multiple issues with reparenting, and made it really hard to start to implement slot-like behavior.
To sum it up, you really want to identify a slot navigator entry as something connected to the conditional itself, and not directly to the root element inside, because conditionals have special behavior. 
With my earlier change several stuff worked out of the box (because reparenting into a clause was similar to reparenting into its root element), but customizing this behavior is much easier with the original code design.

To avoid regression when selecting the clause label in the navigator, I made a new solution: I get the clause element path from the metadata of the conditional. (Introduced a new function called `getConditionalClausePathForNavigatorEntry` for that).

### CODE REVIEW

**Review by ruggi — APPROVED** [review]

File: editor/src/components/navigator/navigator-item/navigator-item.tsx [hunk 1]
```diff
@@ -479,9 +491,27 @@ export const NavigatorItem: React.FunctionComponent<
   )
   const select = React.useCallback(
     (event: any) => {
-      selectItem(dispatch, getSelectedViewsInRange, navigatorEntry, index, selected, event)
+      if (isConditionalClauseNavigatorEntry(navigatorEntry) && elementMetadata != null) {
```
Comment (seanparsons) [inline editor/src/components/navigator/navigator-item/navigator-item.tsx:494]: I think we want this logic inside `selectItem`, as it's now split between this handler and the function.

**Review by seanparsons — APPROVED** [review]

File: editor/src/components/navigator/navigator-utils.ts [hunk 2]
```diff
@@ -187,3 +193,23 @@ export function getNavigatorTargets(
     visibleNavigatorTargets: visibleNavigatorTargets,
   }
 }
+
+export function getConditionalClausePathForNavigatorEntry(
```
[the remark this answers is not in the document]
Comment (Rheeseyb) [reply editor/src/components/navigator/navigator-utils.ts:197]: Agreed 👍 the current name doesn't call out that it is showing the active path, so this rename would help a lot

[the remark this answers is not in the document]
Comment (gbalint) [reply editor/src/components/navigator/navigator-utils.ts:197]: But this function does not return the active clause, but the clause defined in the navigatorEntry object

[the remark this answers is not in the document]
Comment (Rheeseyb) [reply editor/src/components/navigator/navigator-utils.ts:197]: 🤦 yep, you are right

**Review by Rheeseyb — APPROVED** [review]

File: editor/src/components/navigator/navigator-item/navigator-item.tsx (hunk 1 above)
Comment (gbalint) [reply editor/src/components/navigator/navigator-item/navigator-item.tsx]: Good idea, see 32c6cb04f248c748922f47042198f1bb3bf62d16

File: editor/src/components/navigator/navigator-item/navigator-item.tsx [hunk 3]
```diff
@@ -79,21 +78,21 @@ export interface NavigatorItemInnerProps {
 function selectItem(
   dispatch: EditorDispatch,
   getSelectedViewsInRange: (i: number) => Array<ElementPath>,
-  navigatorEntry: NavigatorEntry,
+  elementPath: ElementPath,
```
[the remark this answers is not in the document]
Comment (gbalint) [reply editor/src/components/navigator/navigator-item/navigator-item.tsx]: Yes, by moving the logic inside selectItem I changed back the parameter to NavigatorEntry, see 32c6cb04f248c748922f47042198f1bb3bf62d16

File: editor/src/components/navigator/navigator-utils.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ruggi) [reply editor/src/components/navigator/navigator-utils.ts:197]: mmh true, but imho this still sounds like the navigator entry variant for `getConditionalClausePath`, which returns the path of a given branch, not a derived one 🤔

**Pull request merged**